packages/backend-defaults/report-auditor.api.md

```diff
@@ -0,0 +1,138 @@
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+import type { AuditorCreateEvent } from '@backstage/backend-plugin-api';
+import type { AuditorEventSeverityLevel } from '@backstage/backend-plugin-api';
+import { AuditorService } from '@backstage/backend-plugin-api';
+import type { AuthService } from '@backstage/backend-plugin-api';
+import type { Config } from '@backstage/config';
+import type { Format } from 'logform';
+import type { HttpAuthService } from '@backstage/backend-plugin-api';
+import type { JsonObject } from '@backstage/types';
+import type { PluginMetadataService } from '@backstage/backend-plugin-api';
+import type { Request as Request_2 } from 'express';
+import { ServiceFactory } from '@backstage/backend-plugin-api';
+import * as winston from 'winston';
+
+// @public
+export class Auditor implements AuditorService {
+  // (undocumented)
+  addRedactions(redactions: Iterable<string>): void;
+  // (undocumented)
+  child(
+    meta: JsonObject,
+    deps?: {
+      auth?: AuthService;
+      httpAuth?: HttpAuthService;
+      plugin?: PluginMetadataService;
+    },
+  ): AuditorService;
+  static colorFormat(): Format;
+  static create(options?: AuditorOptions): Auditor;
+  // (undocumented)
+  createEvent<TMeta extends JsonObject>(
+    options: Parameters<AuditorCreateEvent<TMeta>>[0],
+  ): ReturnType<AuditorCreateEvent<TMeta>>;
+  static redacter(): {
+    format: Format;
+    add: (redactions: Iterable<string>) => void;
+  };
+}
+
+// @public
+export type AuditorEvent = [
+  eventId: string,
+  meta: {
+    severityLevel: AuditorEventSeverityLevel;
+    actor: AuditorEventActorDetails;
+    meta?: JsonObject;
+    request?: AuditorEventRequest;
+  } & AuditorEventStatus,
+];
+
+// @public (undocumented)
+export type AuditorEventActorDetails = {
+  actorId?: string;
+  ip?: string;
+  hostname?: string;
+  userAgent?: string;
+};
+
+// @public
+export type AuditorEventOptions<TMeta extends JsonObject> = {
+  eventId: string;
+  severityLevel?: AuditorEventSeverityLevel;
+  request?: Request_2<any, any, any, any, any>;
+  meta?: TMeta;
+} & AuditorEventStatus;
+
+// @public (undocumented)
+export type AuditorEventRequest = {
+  url: string;
+  method: string;
+};
+
+// @public (undocumented)
+export type AuditorEventStatus<TError extends Error = Error> =
+  | {
+      status: 'initiated';
+    }
+  | {
+      status: 'succeeded';
+    }
+  | ({
+      status: 'failed';
+    } & (
+      | {
+          error: TError;
+        }
+      | {
+          errors: TError[];
+        }
+    ));
+
+// @public
+export interface AuditorFactoryOptions {
+  // (undocumented)
+  format: (config?: Config) => winston.Logform.Format;
+  // (undocumented)
+  transports: (config?: Config) => winston.transport[];
+}
+
+// @public
+export const auditorFieldFormat: Format;
+
+// @public (undocumented)
+export interface AuditorOptions {
+  // (undocumented)
+  auth?: AuthService;
```

Comment:
Does it really make sense to have these be optional to begin with? I'm thinking that this will be a new-backend-system-only service